cartridges/app_custom_ylepetchenko/cartridge/models/account.js

```diff
@@ -117,27 +9,15 @@ function getCustomerPaymentInstruments(userPaymentInstruments) {
  * @constructor
  */
 function account(currentCustomer, addressModel, orderModel) {
-    this.profile = getProfile(currentCustomer.raw);
-    this.addresses = getAddresses(currentCustomer.addressBook);
-    this.preferredAddress = addressModel || getPreferredAddress(currentCustomer.addressBook);
-    this.orderHistory = orderModel;
-    this.payment = getPayment(currentCustomer instanceof Customer ? currentCustomer.profile.wallet : currentCustomer.wallet);
-    this.registeredUser = currentCustomer instanceof Customer ? (currentCustomer.authenticated && currentCustomer.registered) : (currentCustomer.raw.authenticated && currentCustomer.raw.registered);
-    this.isExternallyAuthenticated = currentCustomer instanceof Customer ? currentCustomer.externallyAuthenticated : currentCustomer.raw.externallyAuthenticated;
+    base.call(this, currentCustomer, addressModel, orderModel);
 
-    if (currentCustomer instanceof Customer) {
-        this.customerPaymentInstruments = currentCustomer.profile.wallet
-        && currentCustomer.profile.wallet.paymentInstruments
-        ? getCustomerPaymentInstruments(currentCustomer.profile.wallet.paymentInstruments.toArray())
-        : null;
-    } else {
-        this.customerPaymentInstruments = currentCustomer.wallet
-        && currentCustomer.wallet.paymentInstruments
-        ? getCustomerPaymentInstruments(currentCustomer.wallet.paymentInstruments)
-        : null;
+    if(currentCustomer.profile){
```

Comment:
Better to check with empty() function:
if(!empty(currentCustomer.profile)) {}
This function returns true if var or property is null  or undefined